### ISSUE TITLE
Optimize tree view such that cross process calls are batched

### ISSUE THREAD

**Issue by alexr00** [microsoft/vscode#232263]
In the async data tree, `updateChildren` will update the children of any node passed in, even if that node is far, far off screen. 

For the extension tree view, this can result in slowness. Consider the following case:
- Extension contributes a giant tree, with hundreds of thousands of elements at the top level, all expanded
- For each of these elements, we will call `updateChildren`
- This causes hundreds of thousands of round trip calls to the extension host to get the children of each of these top level elements
- In reality, we only need to get the children of the visible elements

A better solution would be to do the `updateChildren` for the visible elements, and for elements who have a visible child, and simply mark the out-of-view elements as needing to have their children updated once they scroll into view. 

Found while fixing https://github.com/microsoft/vscode/issues/230361

**Comment by RedCMD** [comment]
would this affect the find feature?

from my understanding:
The extension host asks the extension for all nodes and their children
the extension is able to provide all 125k+ nodes in less than 100ms

VSCode then takes upwards of 30sec-1min to process those nodes, with ram usage increasing the whole time

does the VSCode host ask the extension host for every node indivually?
could they just be batched processed?

**Comment by alexr00** [comment]
Yes, I expect it would affect find. I experimented with some batching of getting children, but it didn't fit will with the async data tree. I will discuss some with @joaomoreno.

**Comment by alexr00** [comment]
Discussed with @joaomoreno, and we determined that the best bet here is batching at the tree view level, not at the async data tree level. This will significantly reduce the slowness that comes from having a round trip to the extension host for each of those tree items.

I've implemented this, and on my machine, for a tree with 200,000 nodes expanded, empty nodes, at the root, it now takes less than 30 seconds, down from about 2 minutes. Most of this time is spent looping in the async data tree.
